Use get_current_site to look up the site instead of settings.SITE_ID

hedleyroos opened the pull request:
Use `get_current_site` to look up the site instead of `settings.SITE_ID`. This fixes the case where the site is looked up from the request, which is a typical use case for multitenancy systems that don't spin up a Django process per site.

claudep commented:
Thanks for the patch. It would be nice to add some tests for the updated functionality.

hedleyroos commented:
Sure, I'll add tests for the new params.

claudep commented on tests/testapp/tests/test_templatetags.py:
```diff
@@ -105,6 +106,11 @@ def testGetCommentListFromObject(self):
+        # Retrieving a site from the request is not available in Django 1.7,
+        # our minimum supported version.
+        if get_version().startswith("1.7"):
+            return
```
The proper way to skip a test is either to use the unittest.SkipIf/SkipUnless decorators or use self.skipTest if the condition cannot be retrieved at import time.

hedleyroos commented:
I've added the tests.

I didn't add more tests to `views/moderation.py` because the only change is replacing `settings.SITE_ID` with `get_current_site`, and that function itself is already tested in Django.

A bigger issue I picked up is that Django 1.7 cannot lookup the site from the request - the code simply doesn't exist. I added an exception for it in the tests. Any existing installs of Django 1.7 sites will remain unaffected because they will already have SITE_ID set explicitly in their settings, else they wouldn't be able to run at all. Is this acceptable?

claudep commented:
Sure, it's fine. Django 1.7 support will probably be dropped by this package in the next months anyway.

claudep commented on tests/testapp/tests/test_templatetags.py:
```diff
@@ -105,12 +107,10 @@ def testGetCommentListFromObject(self):
 
+    @unittest.skipIf(get_version().startswith("1.7"),
+                    "Retrieving a site from the request is not available in \
+                    Django 1.7")
```
Nice. Could you just remove this awful backslash :-) We are not stuck to a hard 79-chars limit.

hedleyroos replied:
Haha :) Of course.

claudep commented:
Thanks for your contribution!
